### ISSUE TITLE
unexpected MethodNotAllowedException in some PATCH/POST routes

### ISSUE THREAD

**Issue by carcinocron** [MaartenStaa/laravel-41-route-caching#13]
Seems that disabling the routes cache solves this, both by `//comment` and by `TTL=0`.

output of `php artisan routes` is identical.

It would be nice to know:
- if this code was copied from L5 or written for L4.1
- if this code is meant to work with `$router->controller` and `$router->resource`

**Comment by MaartenStaa** [comment]
Hey Michael,

That sounds odd. Have you got a repro case for me to look at, or some more information pointing to where the issue may be?

**Comment by carcinocron** [comment]
I'll try to get back with you on that, I have a pretty tight schedule and my routes files are large, complicated, and NDA'd.

**Comment by MaartenStaa** [comment]
I just noticed the edit of your first comment.
- This code was written originally for Laravel 4.0, and later changed to support 4.1 and 4.2. The code predates Laravel 5.
- Using `$router->resource` is expected to work fully. I just added an integration test that verifies that.
- Using `$router->controller` works partially. The code should succeed in caching these without problems, but since it relies on the file modified time, changing a controller will not invalidate the cache, unless you also modify/touch the routes file.
